File: chat-demo-android/src/main/java/com/twilio/chat/demo/MemberViewHolder.java
package com.twilio.chat.demo;

import com.twilio.chat.demo.R;
import com.twilio.chat.Member;

import android.view.View;
import android.widget.TextView;
import uk.co.ribot.easyadapter.ItemViewHolder;
import uk.co.ribot.easyadapter.PositionInfo;
import uk.co.ribot.easyadapter.annotations.LayoutId;
import uk.co.ribot.easyadapter.annotations.ViewId;

@LayoutId(R.layout.member_item_layout)
public class MemberViewHolder extends ItemViewHolder<Member>
{
    @ViewId(R.id.identity)
    TextView memberIdentity;

    @ViewId(R.id.member_sid)
    TextView memberSid;

    View view;

    public MemberViewHolder(View view)
    {
        super(view);
        this.view = view;
    }

    @Override
    public void onSetListeners()
    {
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v)
            {
                OnMemberClickListener listener = getListener(OnMemberClickListener.class);
                if (listener != null) {
                    listener.onMemberClicked(getItem());
                }
            }
        });
    }

    public interface OnMemberClickListener {
        void onMemberClicked(Member member);
    }

    @Override
    public void onSetValues(Member member, PositionInfo arg1)
    {
        this.memberIdentity.setText(member.getUserInfo().getIdentity());
    }
}

File: chat-demo-android/src/main/java/com/twilio/chat/demo/MemberViewHolder.kt
package com.twilio.chat.demo

import com.twilio.chat.demo.R
import com.twilio.chat.Member

import android.view.View
import android.widget.TextView
import butterknife.BindView
import uk.co.ribot.easyadapter.ItemViewHolder
import uk.co.ribot.easyadapter.PositionInfo
import uk.co.ribot.easyadapter.annotations.LayoutId
import uk.co.ribot.easyadapter.annotations.ViewId

@LayoutId(R.layout.member_item_layout)
class MemberViewHolder(internal var view: View) : ItemViewHolder<Member>(view) {
    internal @BindView(R.id.identity) lateinit var memberIdentity: TextView
    internal @BindView(R.id.member_sid) lateinit var memberSid: TextView

    override fun onSetListeners() {
        view.setOnClickListener {
            val listener = getListener(OnMemberClickListener::class.java)
            listener?.onMemberClicked(item)
        }
    }

    interface OnMemberClickListener {
        fun onMemberClicked(member: Member)
    }

    override fun onSetValues(member: Member, arg1: PositionInfo) {
        memberIdentity.text = member.identity
    }
}
